parquet-hadoop/src/main/java/org/apache/parquet/hadoop/ParquetRecordReader.java

```diff
@@ -155,8 +155,15 @@ private void initializeInternalReader(ParquetInputSplit split, Configuration con
     }
 
     // open a reader with the metadata filter
-    ParquetFileReader reader =
-        ParquetFileReader.open(HadoopInputFile.fromPath(path, configuration), optionsBuilder.build());
+    HadoopInputFile inputFile;
+    if (split.getFooter() != null
+        && split.getFooter().getInputFile() != null
+        && split.getFooter().getInputFile() instanceof HadoopInputFile) {
```

Comment:
cc @amousavigourabi to see if there is any chance to apply this to the non-hadoop code path

Reply:
Hi @wgtmac, given that the `inputFile` variable seems to only be used in a constructor expecting an `InputFile` and not necessarily a `HadoopInputFile`, I think this `instanceof` condition could be dropped. As I just quickly skimmed it now and might have missed something, I'll take a more thorough look after Boxing Day. Happy holidays!🎄🎆

Reply:
Thanks @wgtmac @amousavigourabi  for your review.
 I have changed the `HadoopInputFile` to ``InputFile`

Reply:
Looking good! Going back to @wgtmac's earlier concern, the method this snippet is part of is already in the Hadoop code path and I'm not sure whether there is a more generic alternative available. For the rest, the switch to using the plain `InputFile` interface here is of course amazing for flexibility in the future and makes the code a bit cleaner. Thanks a lot for the swift fix @wankunde!